server/routes/prediction.routes.js

```diff
@@ -359,146 +364,155 @@ function normalizeMatchFilter(filter, field){
  * @return {Promise<Array(Prediction)>} All predictions that match the filter
  */
 /** @namespace filter.numDocs */
-async function getPredictions (filter, user) {
 
-  try {
-    let first = filter.first || 0
-    let max_fetch_rows = filter.rows || configuration.getConfig("defaultMaxPredictions", 1000)
 
+async function getPredictions(filter, user) {
+  try {
+    logger.debug('GetPredictions - Start', { filter, user });
+    let first = filter.first || 0;
+    let max_fetch_rows = filter.rows || configuration.getConfig('defaultMaxPredictions', 1000);
 
-    if ( user === undefined || user.agency === undefined || user.userRole === undefined ) {
-      return []
+    // Ensure user has necessary information
+    if (!user?.agency || !user?.userRole || !user?.email) {
+      logger.warn('Missing user info');
+      return { predictions: [], first: 0, rows: 0, totalCount: 0 };
     }
 
-    logger.debug("Entering getPredictions")
-    // await updatePredictionTable()
+    // Compute the user's office dynamically from their email
+    user.office = getOfficeFromEmail(user.email);
```

Comment:
Shouldn't the user.office already be populated after you login?

Reply:
# Understanding Office Computation in SRT

## Current Flow

### Login (auth.routes.js)
```javascript
// In casStage2:
const office = getOfficeFromEmail(email);
req.session.cas_userinfo.office = office;

// In token creation (convertCASNamesToSRT):
const office = getOfficeFromEmail(srt_userinfo['email'])
srt_userinfo['office'] = office
```

### Predictions (prediction.routes.js)
```javascript
// In getPredictions:
user.office = getOfficeFromEmail(user.email);
```

## Why We Compute Office Again

1. Office is not stored in database
   - We decided to compute it dynamically rather than store it
   - User model doesn't have an office field

2. Token/Session contains office but:
   - We're computing it fresh each time for predictions
   - This ensures we always have the most current mapping

## Current Architecture
- Compute office from email when needed
- Don't persist in database
- Include in session/token info
- Recompute for predictions

So the line, `user.office = getOfficeFromEmail(user.email)` in getPredictions is necessary because we're following the pattern of computing office on demand rather than storing it permanently.

Alternatives could be:
1. Store office in database (decided against this)
2. Extract from token instead of recomputing
3. Keep current approach of computing when needed